app/models/redirection.rb

```diff
@@ -1,5 +1,5 @@
 class Redirection < ActiveRecord::Base
-  belongs_to :next, class_name: "Redirection"
+  belongs_to :next, class_name: "Redirection", optional: true
```

Comment:
Without this line, this broke: https://github.com/gabebw/hotline-webring/blob/12bf532790761a282fbabf30897244af2182b0c1/db/seeds.rb#L2-L6

or this broke:

https://github.com/gabebw/hotline-webring/blob/12bf532790761a282fbabf30897244af2182b0c1/spec/factories.rb#L5

I don't know what changed in Rails 6 that makes this break, since I thought Rails 5 defaulted to `optional: false`, which I'd think would make this break.

Overview:

* Setting `next_id` to a fake value like `-1` breaks the validation here
* I tried `validates :next, presence: true, on: :update` so that creation would at least work, but that broke this:

https://github.com/gabebw/hotline-webring/blob/12bf532790761a282fbabf30897244af2182b0c1/app/models/ring.rb#L12

Basically, we need a way to tell Rails that yes, it's bad data...but we're going to fix it immediately.

Since I don't think we can do that, I just added this setting and will rely on tests.

I'm very open to hearing about a better way to to this!

Reply:
Hmmm yeah that's strange. My research turned up the same thing you found. I don't _love_ it being optional, but I think our test coverage is good enough that it's okay to leave it optional.